packages/expo-screen-orientation/ios/EXScreenOrientation/EXScreenOrientationModule.m

```diff
@@ -0,0 +1,225 @@
+                                           selector:@selector(handleDeviceOrientationChange:)
+                                               name:UIDeviceOrientationDidChangeNotification
+                                             object:[UIDevice currentDevice]];
+  _hasListeners = YES;
+}
+
+// Will be called when this module's last listener is removed, or on dealloc.
+- (void)stopObserving
+{
+  [[NSNotificationCenter defaultCenter] removeObserver:self];
+  _hasListeners = NO;
+}
+
+- (void)handleDeviceOrientationChange:(NSNotification *)notification
+{
+  UIDevice *device = notification.object;
+  UIInterfaceOrientation currentDeviceOrientation = [EXScreenOrientationUtilities UIDeviceOrientationToUIInterfaceOrientation:[device orientation]];
+  UIInterfaceOrientationMask currentOrientationMask = [_screenOrienationRegistry currentOrientationMask];
+  
+  // checks if screen orientation should be changed when user rotates the device
+  if ([EXScreenOrientationUtilities doesOrientationMask:currentOrientationMask containOrientation:currentDeviceOrientation] && _currentScreenOrientation != currentDeviceOrientation) {
+    // updates the current screen orientation
+    _currentScreenOrientation = currentDeviceOrientation;
+    
+    // we sends event only if current screen orientation was changed
+    [self handleScreenOrientationChange];
+  }
+}
+
+- (void)handleScreenOrientationChange
+{
+  if (_hasListeners) {
+    [_eventEmitter sendEventWithName:@"expoDidUpdateDimensions" body:@{
+      @"orientation": [EXScreenOrientationUtilities exportOrientation:_currentScreenOrientation],
+      @"orientationLock": [EXScreenOrientationUtilities exportOrientationLock:[_screenOrienationRegistry currentOrientationMask]]
+    }];
+  }
+}
+
+- (NSArray<NSString *> *)supportedEvents {
+  return @[@"expoDidUpdateDimensions"];
+}
+
+- (void)enforceDesiredDeviceOrientationWithOrientationMask:(UIInterfaceOrientationMask)orientationMask
+{
+  // if current sreen orientation isn't part of the mask, we have to change orientation for default one included in mask, in order up-left-right-down
+  if (![EXScreenOrientationUtilities doesOrientationMask:orientationMask containOrientation:_currentScreenOrientation]) {
+    __block UIInterfaceOrientation newOrientation = [EXScreenOrientationUtilities defaultOrientationForOrientationMask:orientationMask];
+    if (newOrientation != UIInterfaceOrientationUnknown) {
+      UM_WEAKIFY(self);
+      dispatch_async(dispatch_get_main_queue(), ^{
+        UM_STRONGIFY(self);
+        self.currentScreenOrientation = newOrientation;
+        [[UIDevice currentDevice] setValue:@(newOrientation) forKey:@"orientation"];
+        [UIViewController attemptRotationToDeviceOrientation];
+      
+        // screen orientation changed so we send event (notification isn't triggered when manually changing orienatation)
+        [self handleScreenOrientationChange];
+      });
+    }
+  }
+}
+
+- (void)onAppBackgrounded {
+
```

Comment:
Do you think it would make sense to add a `moduleDidBackground` method? Maybe if someone used this module in a brownfield app it would make sense to clear the lock when the React Native "fragment" backgrounds?